sdv/model.py

```diff
@@ -832,9 +832,13 @@ class Service(Node):
 
     def __init__(self):
         super().__init__()
-        _address = conf.service_locator.get_location(self.name)
-        self.channel = grpc.aio.insecure_channel(_address)  # type: ignore
-        self.metadata = conf.service_locator.get_metadata(self.name)
+        service_locator = config.middleware.service_locator
+        _location = service_locator.get_service_location(self.name)
+        _hostname = urlparse(_location).hostname
+        _port = urlparse(_location).port
+        self.address = f"{_hostname}:{_port}"
```

Comment:
Should be checked if `_port` is `None`, otherwise we end up with "<hostname>:" ...

Reply:
related to the one above!